push.sh

```diff
@@ -1,15 +1,59 @@
-#!/bin/bash
+#!/usr/bin/env bash
 #
 # Build all Python images image from the currently checked out version and
 # push them it to the Docker Hub.
 
+set -e -o pipefail -u
+
+image_name='metabrainz/python'
+
+DOCKER_CMD=${DOCKER_CMD:-docker}
+
+# shellcheck disable=SC2207
+remote_tags=($(wget -q \
+	https://registry.hub.docker.com/v1/repositories/${image_name}/tags \
+	-O - | jq -r '.[] | .name'))
+
 for version in 2.7 3.6 3.7 3.8
 do
 	pushd "$(dirname "${BASH_SOURCE[0]}")/${version}/"
 	echo "Building ${version}..."
-	docker build -t metabrainz/python:${version} .
+	${DOCKER_CMD} build -t ${image_name}:${version} .
+	created=$(${DOCKER_CMD} inspect -f '{{.Created}}' ${image_name}:${version} \
+		| sed 's/^\(....\)-\(..\)-\(..\)T.*$/\1\2\3/')
+	date_version=${version}-${created}
+	if [[ " ${remote_tags[*]} " == *" ${date_version} "* ]]
+	then
```

Comment:
it seems like this code exists to do the following:
 - build an image metabrainz/python:3.5
 - find the date that the image was built using docker-inspect
 - retag it to include the date that it was built
 - if this tag already exists, pull that tag, add .0 to the end, and re-push (backup)
 - push the recently built image

I don't think that we're going to have many situations where we build such an image more than once in a day. In fact, we build these images very rarely. 
My plan for this would have been to do either:
 - Automatically get the current date using `date` and add to the `-t` flag of docker build
 - just accept this date as an argument to push.sh, and manage versions manually (as we do in listenbrainz, acousticbrainz)

Reply:
Already discussed on IRC but here is a summary. This pull request is following Docker Hub’s common practices to update stable tag `x.y` while adding more specific tag  `x.y-date`. Added documentation in commit <https://github.com/metabrainz/docker-python/pull/9/commits/483e4af13dd782ace3d8bbe601c019dcf9de4a8f>.

> I don't think that we're going to have many situations where we build such an image more than once in a day.

We already had the case with `3.6-1`. It is rare but it seems safer to have a plan when it happens.

> Automatically get the current date using `date` and add to the `-t` flag of docker build

Yes, that would work too. But the idea here is to push multiple tags anyway.

> just accept this date as an argument to push.sh

I would avoid that as it makes very easy to enter a wrong date or to override an existing tag using shell history.